leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionServiceBuilder.java

```diff
@@ -114,34 +107,51 @@ public PaxosLeaderElectionServiceBuilder eventRecorder(PaxosLeaderElectionEventR
         return this;
     }
 
+    public PaxosLeaderElectionServiceBuilder quorumSize(int quorumSize) {
+        this.quorumSize = quorumSize;
+        return this;
+    }
+
+    public PaxosLeaderElectionServiceBuilder leaderUuid(UUID leaderUuid) {
+        this.leaderUuid = leaderUuid;
+        return this;
+    }
+
+    private int quorumSize() {
+        Preconditions.checkArgument(quorumSize > 0, "quorum size not set");
+        return quorumSize;
+    }
+
+    private UUID leaderUuid() {
+        return Preconditions.checkNotNull(leaderUuid, "leader uuid not set");
+    }
```

Comment:
Hmm, I think it's okay to not check here and do all the checks in `LeaderElectionServiceBuilder` - it seems a bit strange that we do some but not all of them here, and then do a full set of them in LESB. Might have read something incorrectly.

Reply:
sure, I think I started off here, and then moved into a new builder